src/Test.UnitTests.BinSkim.Driver/MultithreadedAnalyzeCommandTests.cs

```diff
@@ -74,9 +116,31 @@ public void MultithreadedAnalyzeCommand_ReturnCommonPathRootFromTargetSpecifiers
                 {
                     sb.AppendLine($"The test was expecting '{testCase.ExpectedCommonPath}' but found '{commonPath}'.");
                 }
+
+                // Testing the same string array in a random order.
+                // This will guarantee that the sorting is working as expected.
+                commonPath = MultithreadedAnalyzeCommand.ReturnCommonPathRootFromTargetSpecifiersIfOneExists(Shuffle(testCase.TargetFileSpecifiers));
+                if (commonPath != testCase.ExpectedCommonPath)
+                {
+                    sb.AppendLine($"The test was expecting '{testCase.ExpectedCommonPath}' but found '{commonPath}' when shuffed.");
+                }
             }
 
             sb.Length.Should().Be(0, sb.ToString());
         }
+
+        private static string[] Shuffle(string[] data)
+        {
+            var rand = new Random();
```

Comment:
>Random [](http://example.com/codeflow?start=27&length=6)

Don't ever introduce randomization into tests without first providing a generated seed to the random class. Next you need to report the seed that was used.

Otherwise you may not easily be able to reproduce a rarely generated error!

Reply:
Go look at how this is done in security utilities.

[https://github.com/microsoft/security-utilities/pull/15/files#diff-8e7029bcfb4b2255f03585b66339672c7bb45d04e8835b4200b82f42740d0460](https://github.com/microsoft/security-utilities/pull/15/files#diff-8e7029bcfb4b2255f03585b66339672c7bb45d04e8835b4200b82f42740d0460)

Reply:
fixed!
thanks for that and added that to my notes :)